app/components/agent/Show.js

```diff
@@ -0,0 +1,267 @@
+        }).isRequired
+      }).isRequired
+    }),
+    relay: React.PropTypes.object.isRequired
+  };
+
+  state = {
+    stopping: false
+  };
+
+  componentDidMount() {
+    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, AGENT_REFRESH_INTERVAL);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this._agentRefreshInterval);
+  }
+
+  fetchUpdatedData = () => {
+    this.props.relay.forceFetch(true);
+  };
+
+  renderExtraItem(title, content) {
+    return (
+      <li key={title}>
+        <strong className="black">{title}:</strong> {content}
+      </li>
+    );
+  }
+
+  renderExtras(agent) {
+    const extras = [];
+
+    if (agent.version) {
+      extras.push(this.renderExtraItem('Version', agent.version));
+    }
+
+    if (agent.hostname) {
+      extras.push(this.renderExtraItem('Hostname', agent.hostname));
+    }
+
+    if (agent.pid) {
+      extras.push(this.renderExtraItem('PID', agent.pid));
+    }
+
+    if (agent.ipAddress) {
+      extras.push(this.renderExtraItem('IP Address', agent.ipAddress));
+    }
+
+    if (agent.userAgent) {
+      extras.push(this.renderExtraItem('User Agent', agent.userAgent));
+    }
+
+    if (agent.operatingSystem) {
+      extras.push(this.renderExtraItem('OS', agent.operatingSystem));
+    }
+
+    if (agent.priority) {
+      extras.push(this.renderExtraItem('Priority', agent.priority));
+    }
+
+    if (agent.job) {
+      extras.push(this.renderExtraItem('Running', <a href="agent.job.path">{agent.job.projectName} - Build {agent.job.buildNumber} / <Emojify>{agent.job.name || agent.job.command}</Emojify></a>));
+    }
+
+    if (agent.connectedAt) {
+      extras.push(this.renderExtraItem(
+        'Connected',
+        <span>
+          <FriendlyTime value={agent.connectedAt} capitalized={true} />
+          {agent.pingedAt && agent.connectionState === 'connected' &&
+            <span> (last check-in was {<FriendlyTime value={agent.pingedAt} />})</span>
+          }
+        </span>
+      ));
+    }
+
+    if (agent.connectionState === 'disconnected') {
+      extras.push(this.renderExtraItem('Disconnected', <FriendlyTime value={agent.disconnectedAt} />));
+    } else if (agent.connectionState === 'lost') {
+      extras.push(this.renderExtraItem('Lost', <FriendlyTime value={agent.lostAt} />));
+    } else if (agent.connectionState === 'stopped' || agent.connectionState === 'stopping') {
+      extras.push(this.renderExtraItem('Stopped', <span><FriendlyTime value={agent.stoppedAt} /> by {agent.stoppedBy.name}</span>));
+
+      // Also show when the agent eventually disconnected
+      if (agent.disconnectedAt) {
+        extras.push(this.renderExtraItem('Disconnected', <FriendlyTime value={agent.disconnectedAt} />));
+      }
+    }
+
+    return extras;
+  }
+
+  handleStopButtonClick = (evt) => {
+    evt.preventDefault();
+
+    this.setState({ stopping: true });
+
+    const mutation = new AgentStopMutation({
+      agent: this.props.agent,
+      graceful: true
```

Comment:
After looking at the current behaviour in the Stop button we do a `graceful: false` so we should do the same here I think!